### PR TITLE
add CloseOnDispose for durable subscription

### PR DESCRIPTION
**Pull request by itn3000**
default behavior is Unsubscribe() for compatibility

### CODE REVIEW

**Comment by sixlettervariables** [comment]
It would be more typical of .Net to call it LeaveOpen, which this could default to true. (Although the default would be opposite most of the framework)

Thoughts?

**Comment by ColinSullivan1** [comment]
@itn3000, Thank you for the PR!

And now the hard part, naming.  :)    

@sixlettervariables , LeaveOpen certainly makes sense, good suggestion.  

I also found an [AutoDisposeNNN](https://msdn.microsoft.com/en-us/library/system.servicemodel.operationbehaviorattribute.autodisposeparameters%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396) , property, which lead me to think of `AutoCloseSubscriber`, which would be disabled by default, but could be enabled.  It is specific enough that it'd leave room for additional options around disposal behavior in the future.

What do you both think?

**Comment by sixlettervariables** [comment]
I would defer to the [guidance for IDisposable implementations](https://msdn.microsoft.com/en-us/library/b1yfkh5e(v=vs.110).aspx) to ensure, "when in Rome," behavior. Usually Dispose and Close are synonymous, and you then provide a means to toggle that behavior away from the default.

> ✓ CONSIDER providing method Close(), in addition to the Dispose(), if close is standard terminology in the area.
> When doing so, it is important that you make the Close implementation identical to Dispose and consider implementing the IDisposable.Dispose method explicitly.

(I'm honestly not familiar enough with the ServiceModel framework to comment on that approach.)

**Comment by ColinSullivan1** [comment]
@sixlettervariables , I agree, and had Close() been supported at the initial release of the client, I probably have included it.  Since it is an addition (https://github.com/nats-io/csharp-nats-streaming/pull/35), and we can't change behavior, I think we'll need to toggle it to enable it. 

I think we should go with for `AutoCloseSubscriber`, disabled by default.

**Comment by ColinSullivan1** [comment]
@itn3000 @sixlettervariables , Admittedly, naming methods and properties is not a strength of mine.  I'm good with `LeaveOpen` if it seems clearer.

Thanks,
Colin

**Comment by itn3000** [comment]
sorry for late response. I fixed the branch.

File: STAN.CLIENT/SubscriptionOptions.cs [hunk 1]
```diff
@@ -96,6 +97,17 @@ public bool ManualAcks
             get { return manualAcks; }
             set { manualAcks = value; }
         }
+        /// <summary>
+        /// Do Close() on Disposing subscription if true, or Unsubscribe(). If you want to resume subscription with durable name, set true.
+        /// </summary>
+        /// <remarks>
+        /// If Close() or Unsubscribe() is called before Disposing, this flag has no affect
```
Comment (ColinSullivan1) [inline STAN.CLIENT/SubscriptionOptions.cs]: Minor typo; "this flag has no effect."

Comment (itn3000) [inline STAN.CLIENT/SubscriptionOptions.cs]: Thank you pointed out. I fixed

**Comment by ColinSullivan1** [comment]
Thank you for the contribution!

**Pull request merged**